posthog/hogql_queries/web_analytics/web_overview_pre_aggregated.py

```diff
@@ -0,0 +1,142 @@
+
+        # Define date filter conditions first
+        current_period_filter = f"day_bucket >= '{current_date_from}' AND day_bucket <= '{current_date_to}'"
+        previous_period_filter = f"day_bucket >= '{previous_date_from}' AND day_bucket <= '{previous_date_to}'"
+
+        # Define common query patterns for averages over sessions
+        def safe_avg_calc(metric, period_filter):
+            sessions = f"uniqMergeIf(sessions_uniq_state, {period_filter})"
+            return f"""
+            if(
+                {sessions} > 0,
+                sumMergeIf({metric}_state, {period_filter}) / {sessions},
+                0
+            )"""
+
+        query_str = f"""
+        SELECT
+            uniqMergeIf(persons_uniq_state, {current_period_filter}) AS unique_persons,
+            uniqMergeIf(persons_uniq_state, {previous_period_filter}) AS previous_unique_persons,
+
+            sumMergeIf(pageviews_count_state, {current_period_filter}) AS pageviews,
+            sumMergeIf(pageviews_count_state, {previous_period_filter}) AS previous_pageviews,
+
+            uniqMergeIf(sessions_uniq_state, {current_period_filter}) AS unique_sessions,
+            uniqMergeIf(sessions_uniq_state, {previous_period_filter}) AS previous_unique_sessions,
+
+            {safe_avg_calc("total_session_duration", current_period_filter)} AS avg_session_duration,
+            {safe_avg_calc("total_session_duration", previous_period_filter)} AS previous_avg_session_duration,
+
+            {safe_avg_calc("total_bounces", current_period_filter)} AS bounce_rate,
+            {safe_avg_calc("total_bounces", previous_period_filter)} AS previous_bounce_rate,
+
+            NULL AS revenue,
+            NULL AS previous_revenue
+        FROM web_overview_daily
+        """
+
+        query = parse_select(query_str)
+
+        filters = self._get_filters()
+        if filters:
+            query.where = filters
+
+        return query
+
+    # We can probably use the hogql general filters somehow but it was not working by default and it was a lot of moving parts to debug at once so
+    # TODO: come back to this later to make sure we're not overcomplicating things
+    def _get_filters(self):
+        """Generate property filters for pre-aggregated tables"""
+        if not self.runner.query.properties:
+            return None
+
+        # Build filter expressions
+        filter_parts = []
+
+        # Only process properties that we know how to map to pre-aggregated tables
+        for posthog_field, table_field in self.SUPPORTED_PROPERTIES.items():
+            for prop in self.runner.query.properties:
+                if hasattr(prop, "key") and prop.key == posthog_field and hasattr(prop, "value"):
+                    value = prop.value
+
+                    # Extract ID if present
+                    if hasattr(value, "id"):
+                        value = value.id
+
+                    # The device_type input differs between "Desktop" | ["Mobile", "Tablet"]
```

Comment:
This is too complicated. Just make it a list at all times (maybe just one entry) and this will be much simpler

Reply:
Good point, the comment was just because that is how our filters already send these values.